extensions/api/data-management/policydefinition/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/policy/service/PolicyServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022 ZF Friedrichshafen AG
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *    ZF Friedrichshafen AG - Initial API and Implementation
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.policy.service;
+
+import org.eclipse.dataspaceconnector.api.result.ServiceResult;
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.spi.contract.offer.store.ContractDefinitionStore;
+import org.eclipse.dataspaceconnector.spi.policy.store.PolicyStore;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+
+import java.util.Collection;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+public class PolicyServiceImpl implements PolicyService {
+
+    private final TransactionContext transactionContext;
+    private final PolicyStore policyStore;
+    private final ContractDefinitionStore contractDefinitionStore;
+
+    public PolicyServiceImpl(TransactionContext transactionContext, PolicyStore policyStore, ContractDefinitionStore contractDefinitionStore) {
+        this.transactionContext = transactionContext;
+        this.policyStore = policyStore;
+        this.contractDefinitionStore = contractDefinitionStore;
+    }
+
+    @Override
+    public Policy findById(String policyId) {
+        return policyStore.findById(policyId);
```

Comment:
please wrap this with the `transactionContext.execute`, same for the `query` method